easybuild/tools/modules.py

```diff
@@ -180,16 +180,18 @@ def __init__(self, mod_paths=None, testing=False):
         if mod_paths is not None:
             self.set_mod_paths(mod_paths)
 
+        cmd_path = which(self.cmd, log_ok=False, log_error=False)
         # only use command path in environment variable if command in not available in $PATH
-        if which(self.cmd) is None and env_cmd_path is not None:
-            self.log.debug("Set %s command via environment variable %s: %s",
-                           self.NAME, self.COMMAND_ENVIRONMENT, self.cmd)
-            self.cmd = env_cmd_path
+        if which(self.cmd) is None:
+            if env_cmd_path is not None:
```

Comment:
Because it was logically wrong. The logic is:
- if command is not in path -> if env variable is given -> use it
- if command is in path -> if env variable is given -> check for consistency

Otherwise you could end up with some strange log message.

Oh I see something else: You only want to run any of the code if the anv variable is given. This makes it easier...